Contribution templates (#3)
Roxxas96/web-bde: CONTRIBUTING.md

# Web BDE : Contributing guide

This document aims at describing the flow of all sub-projects concerning Web BDE.

## Git workflow

The **main** repository [web-bde](https://github.com/Web-BDE/web-bde) is composed of **submodules**, each submodules concern a particular **micro-service** ([front-web-bde](https://github.com/Web-BDE/front-web-bde) for example is the service that distribute static files). Each micro service project has its own Issues/PR and follow **this workflow**.

> In each repository yo wil find 3 types of branch:

Reviewer: ```suggestion
In each repository you will find 3 types of branch:
```

Author: Thank you for the typo !

<type>[optional scope]: <description>

[optional body]

[optional footer(s)]
```

It is composed of :

- The type of the commit (fix, feat, test...)
- An optional scope to link it with a PR/Issue
- A brief description of 50 character maximum
- An optional body describing what and why, but not how ( the purpose, not the implementation )
- An optional footer for additional information like a breaking change

Please be aware of this conventions while committing to your branch. For more information please read the [conventional commits website](https://www.conventionalcommits.org/en/v1.0.0/)

## When and how to make an Issue ?

An Issue has to be opened when a functional needs is discovered, it could be a feature request, bug report, documentation.... You **must** open an Issue before creating a development branch. An issue should not be technical and should describe what the end user will get.

The title of your Issue should describe the topic.

You **must** respect the template provided, tow templates are provided : one for Feature/Documentation request, one for Bug report.

Every Issues must contain a task list to follow the progress of the Issue :

- [x] Write the contributing.md

> ## When an how to make a Pull Request ?

Reviewer: ```suggestion
## When and how to make a Pull Request ?
```

Author: Thank you !
